integration-testing/build.gradle.kts

```diff
@@ -63,8 +67,14 @@ val functionalTest by tasks.registering(Test::class) {
     testClassesDirs = sourceSets["functionalTest"].output.classesDirs
     classpath = sourceSets["functionalTest"].runtimeClasspath
 
-    systemProperties["kotlinVersion"] = kotlin_version
-    systemProperties["atomicfuVersion"] = atomicfu_snapshot_version
+    // the kotlin version used to build the library, which is set in root gradle.properties or overriden by the TC config
+    systemProperties["kotlin.version.integration"] = providers.gradleProperty("kotlin_version").orNull
```

Comment:
Wdyt of using `libs.versions.kotlin` instead of `providers.gradleProperty("kotlin_version").orNull` in such a way we will bind all ways of overridden kotlin version with `systemProperties["kotlin.version.integration"]`?

Reply:
Fixed

Reply:
Mm, for some reason it didn't get resolved that way 👀

Reply:
Sorry it is my mistake, I see you used `libs.versions.kotlin` as I wrote, but it will return `provider` instead of real value, so the right call is `libs.versions.kotlin.get()`.